### PR TITLE
Crossview Editor create and document project structure

### PR DESCRIPTION
**Pull request by adrianchivu**
### Prerequisites

- [X] I have written a descriptive pull-request title
- [X] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [X] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [X] I have provided test coverage for my change (where applicable)

### CODE REVIEW

File: cdp4ime-tests.nunit [hunk 1]
```diff
@@ -26,6 +26,7 @@
 	<assembly path="CDP4Reporting.Tests/bin/Debug/net452/CDP4Reporting.Tests.dll" />
```
Comment (cozminvelciu) [inline cdp4ime-tests.nunit:26]: Assembly entry using tab instead of spaces for alignment.

File: cdp4ime-tests.nunit [hunk 2]
```diff
@@ -26,6 +26,7 @@
 	<assembly path="CDP4Reporting.Tests/bin/Debug/net452/CDP4Reporting.Tests.dll" />
     <assembly path="CDP4Grapher.Tests/bin/Debug/net452/CDP4Grapher.Tests.dll" />
     <assembly path="CDP4UpdateServerDal.Tests/bin/Debug/net452/CDP4UpdateServerDal.Tests.dll" />
+	<assembly path="CDP4CrossViewEditor.Tests/bin/Debug/net452/CDP4CrossViewEditor.Tests.dll" />
```
Comment (cozminvelciu) [inline cdp4ime-tests.nunit:29]: Assembly entry using tab instead of spaces for alignment.

File: cdp4ime-tests.nunit [hunk 3]
```diff
@@ -53,5 +54,6 @@
 	<assembly path="CDP4Reporting.Tests/bin/Release/net452/CDP4Reporting.Tests.dll" />
```
Comment (cozminvelciu) [inline cdp4ime-tests.nunit:54]: Assembly entry using tab instead of spaces for alignment.

File: cdp4ime-tests.nunit [hunk 4]
```diff
@@ -53,5 +54,6 @@
 	<assembly path="CDP4Reporting.Tests/bin/Release/net452/CDP4Reporting.Tests.dll" />
     <assembly path="CDP4Grapher.Tests/bin/Release/net452/CDP4Grapher.Tests.dll" />
     <assembly path="CDP4UpdateServerDal.Tests/bin/Release/net452/CDP4UpdateServerDal.Tests.dll" />
+	<assembly path="CDP4CrossViewEditor.Tests/bin/Release/net452/CDP4CrossViewEditor.Tests.dll" />
```
Comment (cozminvelciu) [inline cdp4ime-tests.nunit:57]: Assembly entry using tab instead of spaces for alignment.

File: CDP4CrossViewEditor/CDP4CrossViewEditor.csproj.DotSettings [hunk 5]
```diff
@@ -0,0 +1,2 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+	<s:Boolean x:Key="/Default/CodeInspection/NamespaceProvider/NamespaceFoldersToSkip/=OfficeRibbon/@EntryIndexedValue">True</s:Boolean></wpf:ResourceDictionary>
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/CDP4CrossViewEditor.csproj.DotSettings:2]: Why is this needed?

File: CDP4CrossViewEditor/CrossViewEditorModule.cs [hunk 6]
```diff
@@ -0,0 +1,116 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="CrossViewEditorModule.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Naron Phou, Patxi Ozkoidi, Alexander van Delft, Mihail Militaru.
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/CrossViewEditorModule.cs:5]: Authors should be updated with a real value for all new files, not copy paste.

File: CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs [hunk 7]
```diff
@@ -0,0 +1,301 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="CrossViewEditorRibbonPart.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Naron Phou, Patxi Ozkoidi, Alexander van Delft, Mihail Militaru.
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs:5]: Authors should be updated with a real value for all new files, not copy paste.

File: CDP4CrossViewEditor.Tests/OfficeRibbon/CrossViewEditorRibbonPartTestFixture.cs [hunk 8]
```diff
@@ -0,0 +1,297 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="CrossViewEditorRibbonPartTestFixture.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Naron Phou, Patxi Ozkoidi, Alexander van Delft, Mihail Militaru.
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor.Tests/OfficeRibbon/CrossViewEditorRibbonPartTestFixture.cs:5]: Authors should be updated with a real value for all new files, not copy paste.

File: CDP4CrossViewEditor.Tests/CrossViewEditorModuleTestFixture.cs [hunk 9]
```diff
@@ -0,0 +1,81 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="CrossViewEditorModuleTestFixture.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Naron Phou, Patxi Ozkoidi, Alexander van Delft, Mihail Militaru.
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor.Tests/CrossViewEditorModuleTestFixture.cs:5]: Authors should be updated with a real value for all new files, not copy paste.

File: CDP4CrossViewEditor.Tests/OfficeRibbon/CrossViewEditorRibbonPartTestFixture.cs [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+            this.ribbonPart.GetContent(RibbonButtonId);
+
+            Assert.DoesNotThrowAsync(async () => await this.ribbonPart.OnAction($"Editor_{iteration.Iid}", iteration.Iid.ToString()));
+
+            CDPMessageBus.Current.SendObjectChangeEvent(iteration, EventKind.Removed);
+            Assert.AreEqual(0, this.ribbonPart.Iterations.Count);
+        }
+
+        /// <summary>
+        /// Create new iteration in order to simulate model openning
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor.Tests/OfficeRibbon/CrossViewEditorRibbonPartTestFixture.cs:279]: Typos. Should be:
> Create a new iteration in order to simulate model opening

**Review by cozminvelciu — CHANGES_REQUESTED** [review]

File: CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+    {
+        /// <summary>
+        /// The NLog logger
+        /// </summary>
+        private static Logger logger = LogManager.GetCurrentClassLogger();
+
+        /// <summary>
+        /// The <see cref="IOfficeApplicationWrapper"/> that provides access to the loaded Office application
+        /// </summary>
+        private readonly IOfficeApplicationWrapper officeApplicationWrapper;
```
Comment (cozminvelciu) [inline CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs:59]: Code smell:
> Remove this unread private field 'officeApplicationWrapper' or refactor the code to use its value.

Is this needed?

File: CDP4CrossViewEditor/CDP4CrossViewEditor.csproj [hunk 12]
```diff
@@ -0,0 +1,327 @@
...
+    <ProjectReference Include="..\CDP4OfficeInfrastructure\CDP4OfficeInfrastructure.csproj">
+      <Private>false</Private>
+      <CopyLocalSatelliteAssemblies>false</CopyLocalSatelliteAssemblies>
+      <ExcludeAssets>runtime</ExcludeAssets>
+    </ProjectReference>
+  </ItemGroup>
+  <ItemGroup>
+    <EmbeddedResource Include="Resources\RibbonXml\CrossViewEditorRibbon.xml" />
+  </ItemGroup>
+  <ItemGroup>
```
Comment (alexatrhea) [inline CDP4CrossViewEditor/CDP4CrossViewEditor.csproj:266]: Kinda weird section to see in this.

File: CDP4CrossViewEditor/CDP4CrossViewEditor.csproj.DotSettings (hunk 5 above)
Comment (alexatrhea) [reply CDP4CrossViewEditor/CDP4CrossViewEditor.csproj.DotSettings:2]: Because there is a weird namespace mispattern but it is actually intentional. I guess this makes sure the user is not bothered by the warning.

File: CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs (hunk 11 above)
Comment (alexatrhea) [reply CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs:59]: leave it for now. It might become relevant when business logic is there.

**Review by alexatrhea — APPROVED** [review]
Approved pending some of @cozminvelciu comments

File: CDP4CrossViewEditor/CDP4CrossViewEditor.csproj.DotSettings (hunk 5 above)
Comment (adrianchivu) [reply CDP4CrossViewEditor/CDP4CrossViewEditor.csproj.DotSettings:2]: It's deliberate. By example: check namespace warning inside BasicRdlRibbonPart.cs file and also CrossViewEditorRibbonPart.cs

File: CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs (hunk 11 above)
Comment (adrianchivu) [reply CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs:59]: Yes. It's needed in the near future.

File: CDP4CrossViewEditor/CDP4CrossViewEditor.csproj (hunk 12 above)
Comment (adrianchivu) [reply CDP4CrossViewEditor/CDP4CrossViewEditor.csproj:266]: See it in all projects/plugins that needs to be loaded into add-in.

Comment (alexatrhea) [reply CDP4CrossViewEditor/CDP4CrossViewEditor.csproj:266]: ok, if it does not cause any issues.

**Review by alexatrhea — APPROVED** [review]

**Review by cozminvelciu — APPROVED** [review]
